conda_smithy/linter/lints.py

```diff
@@ -207,6 +209,20 @@ def lint_sources_should_have_hash(sources_section, lints):
                 "or md5 checksum (sha256 preferably)."
             )
 
+    if recipe_version == 2:
+        # make sure there is no templating involved for the hash and they look alright
+        for source in sources_section:
+            if source.get("sha256"):
+                if not re.match(r"[0-9a-f]{64}", source["sha256"]):
+                    lints.append(
+                        "sha256 checksum must be 64 characters long. Templates are not allowed in the sha256 checksum."
+                    )
+            if source.get("md5"):
+                if not re.match(r"[0-9a-f]{32}", source["md5"]):
+                    lints.append(
+                        "md5 checksum must be 32 characters long. Templates are not allowed in the md5 checksum."
+                    )
+
```

Comment:
We should discuss this in the other PR still and remove it from this one.

Reply:
yup

Reply:
removed from this PR. But we still need a release of `rattler-build-conda-compat` for these changes to go in.